client/app/lib/changeteam/itemview.coffee

```diff
@@ -0,0 +1,65 @@
+kd      = require 'kd'
+JView   = require 'app/jview'
+globals = require 'globals'
+
+module.exports = class ChangeTeamListItem extends kd.ListItemView
+
+  JView.mixin @prototype
+
+  constructor: (options = {}, data) ->
+
+    options.cssClass = kd.utils.curry 'change-team-item', options.cssClass
+
+    super options, data
+
+    team = @getData()
+    { customize, slug } = team
+
+    @logoWrapper = new kd.CustomHTMLView
+      cssClass : 'team-logo-wrapper'
+    if logo = team.customize?.logo
+      @logoWrapper.addSubView new kd.CustomHTMLView
+        tagName    : 'img'
+        attributes :
+          src      : logo
+    else
+      @logoWrapper.addSubView new kd.CustomHTMLView
+        cssClass   : 'default-team-logo'
+
+    { groupsController } = kd.singletons
+    currentGroup = groupsController.getCurrentGroup()
+
+    if currentGroup.slug is slug
+      @actionElement = new kd.CustomHTMLView
+        tagName  : 'span'
+        cssClass : 'current-team-label'
+        partial  : 'Current Team'
+      return
+
+    hostname     = globals.config.domains.main
+    domain       = if slug is 'koding' then hostname else "#{slug}.#{hostname}"
+
+    hasInvitation  = team.invitationCode?
+    actionTitle    = if hasInvitation then 'Join' else 'Switch'
+    actionLink     = if hasInvitation
+    then "//#{domain}/Invitation/#{encodeURIComponent team.invitationCode}"
+    else "//#{domain}/-/loginwithtoken?token=#{team.jwtToken}"
```

Comment:
I assume it should not be a problem since validity duration is pretty long (1 week) and a list of teams is always re-loaded from the server every time user opens `Change Team` modal. On the other side, even if user left `Change Team` modal open for 1 week and then clicked `Switch` button, they would still be landed to the selected group but without auto sign in. In this case they will need to sign in manually but I hope it should not be a big deal for such rare case